### ISSUE TITLE
[openwebnet] Add OpenWebNet Binding (with BUS and radio support)

### ISSUE THREAD

**Issue by mvalla** [openhab/openhab-addons#5195]
Since about 1 year [I have been developing and testing](https://community.openhab.org/t/bticino-openwebnet-new-openhab2-binding-ready-for-testing/43377/) a **new OpenWebNet OH2 binding for BTicino/Legrand home automation systems (MyHOME)**.

The binding has to be considered as an enhancement and replacement of the BTicino v1 binding, which has not been developed/supported anymore.
This is also the first known openHAB 2 binding to support both BUS/SCS and BTicino ZigBee (using USB dongle).

### Distinctive features
- first openHAB 2 binding to support OpenWebNet BUS/SCS gateways and devices
- BUS/SCS + ZigBee (MyHOME radio) support in one single binding
- auto-discovery of:
    - BUS/SCS gateways (via UPnP discovery) and devices
    - BTicino/Legrand ZigBee gateways (USB dongles) and devices
- support for BUS/SCS gateways with text passwords (HMAC authentication), like: MyHOMEServer1, MyHOMEScreen10

These WHOs (MyHOME sub-systems) are supported: WHO=1 Lighting (switches/dimmers), WHO=2 Automation (rollershutters), WHO=4 Thermoregulation (thermostats and sensors), WHO=15/25 CEN/CEN+ (scenario buttons and dry contact / IR sensors), WHO=18 Energy Management.

Several beta releases have been [made available on GitHub](https://github.com/mvalla/openhab2-addons/releases) and [Eclipse Marketplace](https://marketplace.eclipse.org/content/openwebnet-2x-binding-testing).

The binding is currently used in production and tested by 100/200 users (prudent estimate).

### Main Community thread
https://community.openhab.org/t/bticino-openwebnet-new-openhab2-binding-ready-for-testing/43377/

### README file
https://github.com/mvalla/openhab2-addons/blob/openwebnet/addons/binding/org.openhab.binding.openwebnet/README.md

### GitHub repository
https://github.com/mvalla/openhab2-addons

A PR will be submitted soon to be included in 2.5.0.

Massi

**Comment by davidgraeff** [comment]
Nice. Every OH1->OH2 port is welcome and could indeed lead to a fast merge.

* Please make sure you are using the new buildsystem before making the PR.
* Please check all your licenses to contain the most recent license text (especially the "-2019" part).
* Check your author tags. Need to be present in every file with the pattern "name - Initial contribution"
* Have javadoc on every class and important public methods. Don't have useless javadoc like:
  ```
  /**
    * This method assigns value to value
    */
  void setValue(int value) { this.value = value; }
 ```
* Check your logging statements: No useless logging (infront of a thing status update or when a thing handler starts etc, OSGi and the framework knows about it). No "error" level logging. If an exception is part of the context, add it as last argument to the logging statement.

**Comment by mvalla** [comment]
Thank @davidgraeff . I will perform the suggested improvements before submitting PR.
Main change will be to move to new build system, but I have seen good guides from maintainers.

Please note many users have already migrated from the OH1 binding to this one, so hopefully the merge will be really fast.

**Comment by lolodomo** [comment]
What about this other PR ?
https://github.com/openhab/openhab2-addons/pull/3220
Is it a replacement that adds more features ?

**Comment by lolodomo** [comment]
You did not open a PR yet ?

**Comment by mvalla** [comment]
@lolodomo compared to #3220, this one has a quite different design, but most importantly does support BUS gateways/devices, while the other is only for wireless zigbee gateways, AFAIK.

This implementation is currently used by quite few users, who have migrated from BTicino 1.x binding.

I did not open a PR yet because I had to spend a lot of time in the migration to bnd-tools first, and then to solve a malfunctioning IDE (I use Eclipse). 

Right now the situation is a bit better so, time permitting, I can clean up the code a bit and then create a first PR.
My plan is to start with a limited set of devices (WHOs), to make the PR simpler to review (less code), get it approved and then add more devices supported in the following milestones.
